### DISCUSSION TITLE
Have you examples or ideas how to add placeholder to the `DateField`?

### DISCUSSION THREAD

**Discussion by ArtemFedorchuk** [adobe/react-spectrum#4686] [General]
I need to do like this :
<img width="820" alt="Screenshot 2023-06-21 at 19 52 19" src="https://github.com/adobe/react-spectrum/assets/43110065/506a9015-f23d-47ff-8bb4-4183350e0d0b">

Currently, I have like this:
<img width="436" alt="Screenshot 2023-06-21 at 19 54 03" src="https://github.com/adobe/react-spectrum/assets/43110065/c49906f3-fa4a-41ca-a7ff-694ec63ebdeb">
<img width="347" alt="Screenshot 2023-06-21 at 19 59 14" src="https://github.com/adobe/react-spectrum/assets/43110065/19a44578-5466-47bd-bf2c-2ae2855c4db0">
<img width="365" alt="Screenshot 2023-06-21 at 19 59 02" src="https://github.com/adobe/react-spectrum/assets/43110065/dcb75fed-a6b9-4cba-a7cb-0b5c8b20b10e">

Are there ways to add the placeholder if the field has no value or is not focused?
And I want to do opening Calendar if you click anywhere inside the field not only the icon.
The Calendar should be closed if the DateField focus is lost.

**Comment by devongovett** [comment]
A placeholder other than the default one isn't really possible because the individual segments are focusable. If you swapped the segments with plain text, then there would be no way to click on a segment to focus it.

It sounds like you don't really want the date to be keyboard editable though? In that case, you could use a custom Button that triggers a popover with a Calendar inside to select the date, instead of a DateField. Then clicking on that button would open the calendar.

**Reply by ArtemFedorchuk** [reply]
I have an idea how to do it.
We should have two components:
1) Segments
2) Placeholder
Then we should check the `value === null` &&  if the focus on the Wrapper is `false` in this way you can see a placeholder but if you set the focus on the field you can see the segments.
Maybe I'm mistaken, what do you think?

Similar behaviour you can see in the [MUI librari](https://mui.com/x/react-date-pickers/date-picker/) I'd like something like this.
<img width="829" alt="Screenshot 2023-06-22 at 18 37 01" src="https://github.com/adobe/react-spectrum/assets/43110065/617955c5-6488-4da9-9442-582247b2360b">
<img width="843" alt="Screenshot 2023-06-22 at 18 37 10" src="https://github.com/adobe/react-spectrum/assets/43110065/611bdfe5-4c6a-41aa-bd54-be5816322164">

**Reply by snowystinger** [reply]
You could probably accomplish that with what @devongovett outlined above, just when the button gains focus, replace it with the actual DateField and auto focus the first segment. If it's empty and blurs, then place back the button. The whole thing acts like a button on touch for mobile anyways, it automatically pops up the calendar when you click anywhere on it.

One thing to note about that MUI one is that due to the focus/blur nature, I can't seem to autofill it using any of my password managers or chrome form filling.

**Reply by ArtemFedorchuk** [reply]
Thanks a lot
